### PR TITLE
fix panic caused by accessing config after repo closed

### PR DESCRIPTION
**Pull request by whyrusleeping**
fixes https://github.com/ipfs/go-ipfs/issues/1589

I was thinking of caching the config inside the `IpfsNode` to make things a bit cleaner, although that would probably break the repo abstraction (which, for the record, i'm totally okay with).

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

File: cmd/ipfs/daemon.go [hunk 1]
```diff
@@ -204,7 +204,13 @@ func daemonFunc(req cmds.Request, res cmds.Response) {
 		return
 	}
 	if routingOption == routingOptionSupernodeKwd {
-		servers, err := repo.Config().SupernodeRouting.ServerIPFSAddrs()
+		rcfg, err := repo.Config()
```
Comment (rht) [inline cmd/ipfs/daemon.go]: What is the difference between this one and L189?

Comment (whyrusleeping) [inline cmd/ipfs/daemon.go]: it would appear nothing. good catch

File: repo/fsrepo/fsrepo.go [hunk 2]
```diff
@@ -460,9 +457,9 @@ func (r *FSRepo) Config() *config.Config {
 	defer packageLock.Unlock()
 
 	if r.closed {
-		panic("repo is closed")
```
Comment (jbenet) [inline repo/fsrepo/fsrepo.go]: the other ops return an error. i'm don't this should be panic. all sort of code all over the place calls `repo.Config()`

**Comment by jbenet** [comment]
@whyrusleeping you removed sadhack. undo that

**Comment by whyrusleeping** [comment]
@jbenet whoops. i keep having to do that to run the tests

**Comment by whyrusleeping** [comment]
@jbenet sadhack is back.

**Pull request merged**